### PR TITLE
fix: The help icons are not aligned with the text in the 'Create Project' flows 

### PR DESCRIPTION
**Pull request by jayanand05**
Closes: #6549

### Description:
fixed icons alignment with the text in the 'Create Project' flows
0303F084-A633-4995-83D7-911EF24D9E55_4_5005_c

### CODE REVIEW

File: packages/frontend/src/components/ProjectConnection/index.tsx [hunk 1]
```diff
@@ -90,9 +90,13 @@ const ProjectForm: FC<Props> = ({
...
-                    <LeftPanelTitle>
-                        <H5>Warehouse connection</H5>
-                        <DocumentationHelpButton href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project#warehouse-connection" />
+                    <LeftPanelTitle
+                        style={{ display: 'flex', alignItems: 'center' }}
+                    >
+                        <Title order={5} style={{ marginRight: '0px' }}>
+                            Warehouse connection
+                        </Title>
+                        <DocumentationHelpButton />
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectConnection/index.tsx:99]: This is great, but we could replace `LeftPanelTitle` with `Mantine`'s one, so: 

```tsx
 <Flex align="center">
                        <Title order={5} style={{ marginRight: '0px' }}>
                            Warehouse connection
                        </Title>
                        <DocumentationHelpButton />
                    </Flex>
```

File: packages/frontend/src/components/ProjectConnection/index.tsx [hunk 2]
```diff
@@ -117,9 +121,13 @@ const ProjectForm: FC<Props> = ({
...
-                    <LeftPanelTitle>
-                        <H5>dbt connection</H5>
-                        <DocumentationHelpButton href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project" />
+                    <LeftPanelTitle
+                        style={{ display: 'flex', alignItems: 'center' }}
+                    >
+                        <Title order={5} style={{ marginRight: '0px' }}>
+                            dbt connection
+                        </Title>
+                        <DocumentationHelpButton />
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectConnection/index.tsx:130]: and same here! 

```
<Flex align="center">
                        <Title order={5} style={{ marginRight: '0px' }}>
                            dbt connection
                        </Title>
                        <DocumentationHelpButton />
                    </Flex>
```

**Review by joaoviana — CHANGES_REQUESTED** [review]
Thanks for this contribution ⭐ 

If you end up going with my suggestions, please delete: 
export const LeftPanelTitle = styled.div`
    margin-bottom: 10px;
    margin-top: 10px;

    h5 {
        display: inline;
        margin-right: 5px;
    }
`;

in `packages/frontend/src/components/ProjectConnection/ProjectConnection.styles.tsx`

Thanks!

File: packages/frontend/src/components/ProjectConnection/index.tsx (hunk 1 above)
Comment (jayanand05) [reply packages/frontend/src/components/ProjectConnection/index.tsx:99]: Thanks for the review @joaoviana , I will do it.

**Comment by jayanand05** [comment]
Thanks for the review @joaoviana , I did the necessary changes.

File: packages/frontend/src/components/ProjectConnection/index.tsx [hunk 3]
```diff
@@ -92,10 +91,12 @@ const ProjectForm: FC<Props> = ({
...
                     {warehouse && getWarehouseIcon(warehouse)}
-                    <LeftPanelTitle>
-                        <H5>Warehouse connection</H5>
-                        <DocumentationHelpButton href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project#warehouse-connection" />
-                    </LeftPanelTitle>
+                    <Flex align="center">
+                        <Title order={5} style={{ marginRight: '0px' }}>
+                            Warehouse connection
+                        </Title>
+                        <DocumentationHelpButton />
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectConnection/index.tsx:98]: looking great! one last thing, let's change keep the href props here and add back the href prop in `DocumentationHelpButton` pls 🙏

File: packages/frontend/src/components/ProjectConnection/index.tsx [hunk 4]
```diff
@@ -119,10 +120,12 @@ const ProjectForm: FC<Props> = ({
...
 
-                    <LeftPanelTitle>
-                        <H5>dbt connection</H5>
-                        <DocumentationHelpButton href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project" />
-                    </LeftPanelTitle>
+                    <Flex align="center">
+                        <Title order={5} style={{ marginRight: '0px' }}>
+                            dbt connection
+                        </Title>
+                        <DocumentationHelpButton />
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectConnection/index.tsx:127]: same here ^

**Review by joaoviana — CHANGES_REQUESTED** [review]
looking gooood! 
2 more changes pls 👍

File: packages/frontend/src/components/ProjectConnection/index.tsx (hunk 3 above)
Comment (jayanand05) [reply packages/frontend/src/components/ProjectConnection/index.tsx:98]: Ok just wanted to clarify that when i keep the href in DocumentationHelpButton it will still be dependent on ActionIcon defined in DocumentationHelpButton file is it fine?

File: packages/frontend/src/components/DocumentationHelpButton.tsx [hunk 5]
```diff
@@ -9,32 +9,27 @@ type Props = React.AnchorHTMLAttributes<HTMLAnchorElement> &
...
     >
-        <Anchor
-            role="button"
+        <ActionIcon
+            component="a"
             target="_blank"
             rel="noreferrer"
-            color="dimmed"
-            {...anchorProps}
+            href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project"
```
Comment (joaoviana) [inline packages/frontend/src/components/DocumentationHelpButton.tsx:24]: I would remove this, and spread `{...anchorProps}` like it was before, so like this: 
```
<ActionIcon
            component="a"
            target="_blank"
            rel="noreferrer"
            {...anchorProps}
            >
            ```
            
            and ensure that above you destruct these:
            
            ```
const DocumentationHelpButton: FC<Props> = ({ iconProps, tooltipProps, ...anchorProps }) => (
            ```

**Review by joaoviana — CHANGES_REQUESTED** [review]
Hopefully this last comment clarifies. 

tldr: 
* `href` shouldn't be hardcoded on `DocumentationHelpButton` as it's reused in a lot of places
* `anchorProps` should be there and spread on the `ActionIcon` 

thanks!

File: packages/frontend/src/components/ProjectConnection/index.tsx [hunk 6]
```diff
@@ -92,10 +91,12 @@ const ProjectForm: FC<Props> = ({
...
                     {warehouse && getWarehouseIcon(warehouse)}
-                    <LeftPanelTitle>
-                        <H5>Warehouse connection</H5>
-                        <DocumentationHelpButton href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project#warehouse-connection" />
-                    </LeftPanelTitle>
+                    <Flex align="center">
+                        <Title order={5} style={{ marginRight: '0px' }}>
+                            Warehouse connection
+                        </Title>
+                        <DocumentationHelpButton href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project" />
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectConnection/index.tsx:98]: the href is different here, pls change to : 
`href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project#warehouse-connection"`

File: packages/frontend/src/components/ProjectConnection/index.tsx (hunk 3 above)
Comment (joaoviana) [reply packages/frontend/src/components/ProjectConnection/index.tsx]: Hopefully https://github.com/lightdash/lightdash/pull/6583#discussion_r1284128674 makes sense @jayanand05 ! 

and also this: https://github.com/lightdash/lightdash/pull/6583#discussion_r1284130972

**Comment by jayanand05** [comment]
Hi @joaoviana, did the final change thanks a lot for the clarity.

**Review by joaoviana — CHANGES_REQUESTED** [review]
Apologies for the back-and-forth on this @jayanand05, but this change on the `DocumentationHelpButton` would affect the positioning of the icons on several places which basically increases the scope of your contribution. 

We can do this in two ways: 

* keep `DocumentationHelpButton` intact: 

here is the raw change of what it was before: 
```tsx
import { Anchor, AnchorProps, Tooltip, TooltipProps } from '@mantine/core';
import { IconHelpCircle } from '@tabler/icons-react';
import { FC } from 'react';
import MantineIcon, { MantineIconProps } from './common/MantineIcon';

type Props = React.AnchorHTMLAttributes<HTMLAnchorElement> &
    Partial<AnchorProps> & {
        iconProps?: Partial<MantineIconProps>;
        tooltipProps?: Partial<TooltipProps>;
    };

const DocumentationHelpButton: FC<Props> = ({
    iconProps,
    tooltipProps,
    ...anchorProps
}) => (
    <Tooltip
        withinPortal
        label="Open documentation"
        position="top"
        maw={350}
        {...tooltipProps}
    >
        <Anchor
            role="button"
            target="_blank"
            rel="noreferrer"
            color="dimmed"
            {...anchorProps}
        >
            <MantineIcon
                icon={IconHelpCircle}
                size="md"
                display="inline"
                {...iconProps}
            />
        </Anchor>
    </Tooltip>
);

export default DocumentationHelpButton;

```

And update the `ProjectConnection/index.tsx` to: 

```tsx
<Flex align="center" gap={2}>
                        <Title order={5}>Warehouse connection</Title>
                        <DocumentationHelpButton
                            href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project#warehouse-connection"
                            pos="relative"
                            top="2px"
                        />
                    </Flex>
```

and

```tsx
<Flex align="center" gap={2}>
                        <Title order={5}>dbt connection</Title>
                        <DocumentationHelpButton
                            href="https://docs.lightdash.com/get-started/setup-lightdash/connect-project"
                            pos="relative"
                            top="2px"
                        />
                    </Flex>
```

or 

Ensure the icon is centred everywhere 

happy to go for option 1 , which I think will take you less time. What do you think?

**Comment by jayanand05** [comment]
Sure @joaoviana, I will do this changes.
